### PR TITLE
check if long press has been added already

### PR DESCRIPTION
**Pull request by qinweigong**
@aaustin

### CODE REVIEW

File: Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -208,7 +203,7 @@
 				ORGANIZATIONNAME = "Branch Metrics";
 				TargetAttributes = {
 					6700165F1940F51400A9E103 = {
-						DevelopmentTeam = 6B2DFWM843;
```
Comment (shortstuffsushi) [inline Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj]: Probably don't want to change this.

File: Branch-SDK/Branch-SDK/Branch.m [hunk 2]
```diff
@@ -1419,4 +1427,78 @@ - (void)serverCallback:(BNCServerResponse *)response {
...
+
+- (void)bnc_addCancelDebugGestureRecognizer {
+    [self bnc_addGesterRecognizer:@selector(bnc_endDebug:)];
+}
+
+- (void)bnc_addGesterRecognizer:(SEL)action {
+    BNCLongPress = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:action];
+    BNCLongPress.cancelsTouchesInView = NO;
+    BNCLongPress.minimumPressDuration = BNCDebugTriggerDuration;
+    if (![BNCSystemObserver isSimulator]) {
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: If you're making use of both conditions here (is and isn't), why check "not" simulator? Just put the simulator scenario first.

File: Branch-SDK/Branch-SDK/Branch.m [hunk 3]
```diff
@@ -835,6 +841,8 @@ - (void)applicationDidBecomeActive {
             }
         });
     }
+    
+    [self bnc_addDebugGestureRecognizer];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: You'll need a corresponding remove in the `applicationWillResignActive` to avoid multiple gestureRecognizers being added.

File: Branch-SDK/Branch-SDK/Branch.m [hunk 4]
```diff
@@ -1419,4 +1427,78 @@ - (void)serverCallback:(BNCServerResponse *)response {
...
+- (void)bnc_addDebugGestureRecognizer {
+    [self bnc_addGesterRecognizer:@selector(bnc_connectToDebug:)];
+}
+
+- (void)bnc_addCancelDebugGestureRecognizer {
+    [self bnc_addGesterRecognizer:@selector(bnc_endDebug:)];
+}
+
+- (void)bnc_addGesterRecognizer:(SEL)action {
+    BNCLongPress = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:action];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: I'm not 100% sure how this works, but if you're adding this to the window, I believe that other gesture recognizers added by the app developers will be ignored. I'm not sure if the "cancelsTouchesInView" will pass through and call both yours and theirs or not.

Comment (aaustin) [inline Branch-SDK/Branch-SDK/Branch.m]: Maybe Qinwei can chime in here, but I believe he found this to be used by Mixpanel for their debugging option

File: Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj (hunk 1 above)
Comment (qinweigong) [inline Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj]: This is changed by Xcode. How to prevent it from being changed?

Comment (shortstuffsushi) [inline Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj]: You can't prevent Xcode from changing it (unless you have that team available in Xcode), but you can certainly just not commit ;)

Comment (qinweigong) [inline Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj]: What's the consequence of having this changed and committed? We've never paid attention to this value, and always let it be so far.

Comment (shortstuffsushi) [inline Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj]: If you guys don't have one specific team that you use, then it's not a big deal. At that point, just a churn things. I usually try not to commit things that are not related to the pull request. In this case, the team isn't related to anything, and probably shouldn't be committed at all, or could be committed as "None" to prevent this churn.

File: Branch-SDK/Branch-SDK/Branch.m (hunk 4 above)
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: From the documentation: If a gesture recognizer doesn’t recognize its gesture or if the value of this property is NO, the view receives all touches in the multi-touch sequence.

File: Branch-SDK/Branch-SDK/Branch.m (hunk 3 above)
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: added

File: Branch-SDK/Branch-SDK/Branch.m (hunk 4 above)
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: Good catch! Replaced with gesture recognizer delegate method shouldRecognizeSimultaneouslyWithGestureRecognizer.

**Comment by aaustin** [comment]
:+1:
